src/operator/gitea.ts

```diff
@@ -0,0 +1,161 @@
+import Operator from '@dot-i/k8s-operator'
+import * as k8s from '@kubernetes/client-node'
+import { KubeConfig } from '@kubernetes/client-node'
+import stream from 'stream'
+
+interface ConditionCheckResult {
+  ready: boolean
+  pod: k8s.V1Pod | undefined
+}
+
+const kc = new KubeConfig()
+// loadFromCluster when deploying on cluster
+// loadFromDefault when locally connecting to cluster
+kc.loadFromCluster()
+const k8sApi = kc.makeApiClient(k8s.CoreV1Api)
+
+function buildTeamString(teamNames: any[]): string {
+  if (teamNames === undefined) return '{}'
+  let result = '{'
+
+  for (let i = 0; i < teamNames.length; i++) {
+    const teamName = teamNames[i]
+    const teamObject = `"/${teamName}":{"otomi":["otomi-viewer", "${teamName}"]}`
+
+    if (i === teamNames.length - 1) {
+      // If it's the last team name, don't add a comma
+      result += teamObject
+    } else {
+      result += `${teamObject}, `
+    }
+  }
+
+  result += '}'
+  return result
+}
+
+async function execGiteaCLICommand(object: k8s.V1Pod) {
+  try {
+    console.debug('Finding namespaces')
+    let namespaces: any
+    try {
+      namespaces = (await k8sApi.listNamespace()).body
+    } catch (error) {
+      console.debug('No namespaces found, exited with error:', error)
+    }
+    console.debug('Filtering namespaces with "team-" prefix')
+    let teamNamespaces: any
+    try {
+      teamNamespaces = namespaces.items
+        .map((namespace) => namespace.metadata?.name)
+        .filter((name) => name && name.startsWith('team-') && name !== 'team-admin')
```

Comment:
This is of course fine BUT it is better to you label.
Every team NS has a label `type: team`

Reply:
I'll see if I can make it work, maybe I can also look for namespaces with that tag using the k8sapi.